function(error) {
                // TODO: failure case
            }